sw/device/tests/alert_handler_lpg_reset_toggle.c

```diff
@@ -0,0 +1,548 @@
+                                         kDifAlertHandlerClassB};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = array_idx,
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = ping_timeout,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));
+}
+
+/**
+ * A utility function to wait enough until the alert handler pings a peripheral
+ * alert
+ */
+void wait_enough_for_alert_ping() {
+  // wait enough
+  if (kDeviceType == kDeviceFpgaCw310) {
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**16)*(400ns) = 12.8s
+    busy_spin_micros(1000 * 12800);
+  } else if (kDeviceType == kDeviceSimDV) {
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**3)*(40ns) = 160us
+    busy_spin_micros(160);
+  } else {
+    // Verilator
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**16)*(8us) = 256s
+    // This seems to be impractical for the current clock frequency config
+    // of the Verilator tests (kClockFreqPeripheralHz = 125K).
+    LOG_FATAL("SUPPORTED PLATFORMS: DV and FPGA");
+    LOG_FATAL("TO SUPPORT THE PLATFORM %d, COMPUTE THE RIGHT WAIT-TIME",
+              kDeviceType);
+  }
+}
+
+/**
+ * Get `num` distinct random numbers in the range [0, `max`] from
+ * RV_CORE_IBEX_RND_DATA.
+ *
+ * @param ibex The Ibex DIF object.
+ * @param num The number of random numbers to get.
+ * @param[out] rnd_buf Pointer to the buffer to write the random numbers to.
+ * @param max The maximum random value returned.
+ */
+static void get_rand_words(dif_rv_core_ibex_t *ibex, int num, uint32_t *rnd_buf,
+                           uint32_t max) {
+  uint32_t rnd_word;
+  for (int i = 0; i < num; ++i) {
+    bool found = false;
+    while (found == false) {
+      // Get a new random number.
+      CHECK_DIF_OK(dif_rv_core_ibex_read_rnd_data(ibex, &rnd_word));
+      rnd_word = rnd_word % max;
+      // Check if the number is unique.
+      found = true;
+      for (int j = 0; j < i; ++j) {
+        if (rnd_buf[j] == rnd_word) {
+          // Start over.
+          found = false;
+          break;
+        }
+      }
+    }
+    // Add the number to the buffer.
+    rnd_buf[i] = rnd_word;
+  }
+}
+
+enum {
+  // Non-volatile counter for the test steps on FPGA.
+  kCounterTestSteps = 0,
+};
```

Comment:
Normally we use the anonymous enum format for constants, not for values that are expected to change. From the rest of the code it seems like this value changes, so it's a bit confusing to see the anonymous-enum pattern here. Maybe a global variable would be better? Or is this format required by the `flash_ctrl_testutils` functions?

Reply:
Thanks Jade, that is a good catch. I think my comment was a little misleading.

In `flash_ctrl_testutils`, there are are 3 non-volatile counter arrays, and I am using `kCounterTestSteps` as a constant ID for the non-volatile counter, counter0.  I updated the comment to reflect this. Hope that addresses your concern.

I keep track of the cunter0's value using the variable `test_step_cnt` that I increment after testing each peripheral..